terraform/instances.tf

```diff
@@ -615,35 +615,12 @@ resource "aws_instance" "worker" {
 ////////////////////////////////
 // Template Files
 
-data "template_file" "hab_sup" {
+data "template_file" "sup_service" {
   template = "${file("${path.module}/templates/hab-sup.service")}"
 
   vars {
-    flags               = "--auto-update --channel ${var.release_channel} --events hab-eventsrv.default --listen-gossip 0.0.0.0:${var.gossip_listen_port} --listen-http 0.0.0.0:${var.http_listen_port}"
-    gossip_listen_port  = "${var.gossip_listen_port}"
-    peer_ip             = "${aws_instance.datastore.0.private_ip}"
-    log_level           = "${var.log_level}"
-  }
-}
-
-data "template_file" "hab_sup_permanent" {
-  template = "${file("${path.module}/templates/hab-sup.service")}"
-
-  vars {
-    flags               = "--auto-update --channel ${var.release_channel} --events hab-eventsrv.default --listen-gossip 0.0.0.0:${var.gossip_listen_port} --listen-http 0.0.0.0:${var.http_listen_port} --permanent-peer"
-    gossip_listen_port  = "${var.gossip_listen_port}"
-    peer_ip             = "${aws_instance.datastore.0.private_ip}"
-    log_level           = "${var.log_level}"
-  }
-}
-
-data "template_file" "hab_sup_seed" {
-  template = "${file("${path.module}/templates/hab-sup.service")}"
-
-  vars {
-    flags               = "--auto-update --channel ${var.release_channel} --events hab-eventsrv.default --listen-gossip 0.0.0.0:${var.gossip_listen_port} --listen-http 0.0.0.0:${var.http_listen_port} --permanent-peer"
-    gossip_listen_port  = "${var.gossip_listen_port}"
-    peer_ip             = "127.0.0.1"
-    log_level           = "${var.log_level}"
+    flags     = "--auto-update --channel ${var.release_channel} --events hab-eventsrv.default --listen-gossip 0.0.0.0:${var.gossip_listen_port} --listen-http 0.0.0.0:${var.http_listen_port}"
+    peers     = "${var.peers}"
```

Comment:
Also, what's the reasoning behind having `peers` be a separate template variable, when it effectively ends up being prepended to the value of `flags` anyway? Why not just set it all up here?

Also also, does this blow up in the template if the variable is empty?